#Taller Econometria 

*   Natalia Forero Avila
*   Karen Acosta Enciso 



1. Instale la librería wooldridge e importe numpy y pandas.

In [ ]:
!pip install wooldridge 
import pandas as pd
import numpy as np
import wooldridge as wd

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.1/5.1 MB 42.5 MB/s eta 0:00:00


2.Cargue el dataset wage1 y guárdelo en una variable. Conserve únicamente las columnas exper, educ y wage

In [ ]:
df=wd.data("wage1")
df

wage  educ  exper  tenure  nonwhite  female  married  numdep  smsa  \
0     3.10    11      2       0         0       1        0       2     1   
1     3.24    12     22       2         0       1        1       3     1   
2     3.00    11      2       0         0       0        0       2     0   
3     6.00     8     44      28         0       0        1       0     1   
4     5.30    12      7       2         0       0        1       1     0   
..     ...   ...    ...     ...       ...     ...      ...     ...   ...   
521  15.00    16     14       2         0       1        1       2     0   
522   2.27    10      2       0         0       1        0       3     0   
523   4.67    15     13      18         0       0        1       3     0   
524  11.56    16      5       1         0       0        1       0     0   
525   3.50    14      5       4         1       1        0       2     0   

     northcen  ...  trcommpu  trade  services  profserv  profocc  clerocc  \
0           0  ...         0      0         0         0        0        0   
1           0  ...         0      0         1         0        0        0   
2           0  ...         0      1         0         0        0        0   
3           0  ...         0      0         0         0        0        1   
4           0  ...         0      0         0         0        0        0   
..        ...  ...       ...    ...       ...       ...      ...      ...   
521         0  ...         0      0         0         1        1        0   
522         0  ...         0      1         0         0        1        0   
523         0  ...         0      0         0         0        1        0   
524         0  ...         0      0         0         0        0        0   
525         0  ...         0      0         0         1        0        1   

     servocc     lwage  expersq  tenursq  
0          0  1.131402        4        0  
1          1  1.175573      484        4  
2          0  1.098612        4        0  
3          0  1.791759     1936      784  
4          0  1.667707       49        4  
..       ...       ...      ...      ...  
521        0  2.708050      196        4  
522        0  0.819780        4        0  
523        0  1.541159      169      324  
524        0  2.447551       25        1  
525        0  1.252763       25       16  

[526 rows x 24 columns]

In [ ]:
dv=df[["wage","educ","exper"]] 
dv

wage  educ  exper
0     3.10    11      2
1     3.24    12     22
2     3.00    11      2
3     6.00     8     44
4     5.30    12      7
..     ...   ...    ...
521  15.00    16     14
522   2.27    10      2
523   4.67    15     13
524  11.56    16      5
525   3.50    14      5

[526 rows x 3 columns]

In [ ]:
Y=dv["wage"]
X=dv[["educ","exper"]] 

3. Pase el dataframe a un array de numpy.

In [ ]:
array1=Y.to_numpy()
array1

array([ 3.0999999 ,  3.24000001,  3.        ,  6.        ,  5.30000019,
        8.75      , 11.25      ,  5.        ,  3.5999999 , 18.18000031,
        6.25      ,  8.13000011,  8.77000046,  5.5       , 22.20000076,
       17.32999992,  7.5       , 10.63000011,  3.5999999 ,  4.5       ,
        6.88000011,  8.47999954,  6.32999992,  0.52999997,  6.        ,
        9.56000042,  7.78000021, 12.5       , 12.5       ,  3.25      ,
       13.        ,  4.5       ,  9.68000031,  5.        ,  4.67999983,
        4.26999998,  6.1500001 ,  3.50999999,  3.        ,  6.25      ,
        7.80999994, 10.        ,  4.5       ,  4.        ,  6.38000011,
       13.69999981,  1.66999996,  2.93000007,  3.6500001 ,  2.9000001 ,
        1.63      ,  8.60000038,  5.        ,  6.        ,  2.5       ,
        3.25      ,  3.4000001 , 10.        , 21.62999916,  4.38000011,
       11.71000004, 12.39000034,  6.25      ,  3.71000004,  7.78000021,
       19.97999954,  6.25      , 10.        ,  5.71000004,  2.  

In [ ]:
array2=X.to_numpy()
array2

array([[11,  2],
       [12, 22],
       [11,  2],
       ...,
       [15, 13],
       [16,  5],
       [14,  5]])

In [ ]:
x=np.c_[np.ones(len(array2)),array2]
x


array([[ 1., 11.,  2.],
       [ 1., 12., 22.],
       [ 1., 11.,  2.],
       ...,
       [ 1., 15., 13.],
       [ 1., 16.,  5.],
       [ 1., 14.,  5.]])

4. Sin ayuda de otra librería diferente a numpy:

A. Calcule los coeficientes

In [ ]:
B=np.linalg.inv(x.T.dot(x)).dot(x.T.dot(Y))
B



array([-3.39053946,  0.64427208,  0.0700954 ])

In [ ]:
Estimado=x@B
Estimado

array([ 3.83664416,  5.88282415,  3.83664416,  4.84783456,  4.83139322,
        7.54867231,  9.25778884,  4.69120242,  6.16320574,  9.10418453,
        7.47857691,  5.19528371,  5.39215638,  5.60244257,  6.51368272,
        7.89914929,  5.0416794 ,  6.10652385,  5.25196559,  6.8641597 ,
        5.1117748 ,  6.37349193,  7.54867231,  4.55101163,  6.28998302,
        7.12809993,  7.6888631 ,  9.09077102,  8.37640355,  2.39449571,
        7.2414637 ,  5.76946039,  6.10652385,  4.83139322,  4.55101163,
        8.45991246,  5.39215638,  1.92026928,  5.83955578,  5.1117748 ,
        4.55101163,  5.74263336,  6.75079593,  6.85074619,  5.75604687,
        7.67544959,  3.12227669,  6.8641597 ,  6.26012816,  5.39215638,
        5.60244257,  7.12809993,  5.39215638,  4.83139322,  4.48091624,
        6.48382786,  6.98790914,  2.67487729,  8.76712106,  7.40848152,
        7.78881336,  8.57327622,  5.57561555,  3.96342144,  6.26012816,
        7.45174989,  7.40848152,  6.09311034,  7.61876771,  4.55

In [ ]:
error=Y-Estimado 
error

0     -0.736644
1     -2.642824
2     -0.836644
3      1.152165
4      0.468607
         ...   
521    7.100851
522   -0.922372
523   -2.514782
524    4.291710
525   -2.479747
Name: wage, Length: 526, dtype: float64

In [ ]:
Var= sum(error**2)/523 
Var

10.608336116213543

In [ ]:
x.T @ x

array([[   526.,   6608.,   8951.],
       [  6608.,  87040., 106539.],
       [  8951., 106539., 249027.]])

In [ ]:
f= x.T.dot(x)
inverF= np.linalg.inv(f)
inverF

array([[ 5.53926242e-02, -3.71227943e-03, -4.02835200e-04],
       [-3.71227943e-03,  2.72907389e-04,  1.66782433e-05],
       [-4.02835200e-04,  1.66782433e-05,  1.13597903e-05]])

B. Calcule los errores estándar

In [ ]:
error_est= np.diag(np.sqrt(Var*inverF))
error_est

<ipython-input-42-3d601b1e16fd>:1: RuntimeWarning: invalid value encountered in sqrt
  error_est= np.diag(np.sqrt(Var*inverF))


array([0.76656609, 0.05380607, 0.01097764])

C. Calcule los estadísticos t

In [ ]:
T=B/error_est
T

array([-4.423023  , 11.97396614,  6.38529107])

D. Calcule los intervalos de confianza

In [ ]:
IC= B- 1.96*error_est
IC

array([-4.89300901,  0.53881218,  0.04857923])

In [ ]:
IC2=B+1.96*error_est
IC2

array([-1.88806992,  0.74973198,  0.09161156])

5. Interprete los coeficientes y determine la significancia estadística.


Y^ = -3.3905^ + 0.6443 educ^ +0.0701 exper^

Entendiendo que el modelo es LIN LIN, se puede interpretar que:

a) Educ: Un incremento en un año de educación tiene un efecto de aumento en el salario de 0.6443 dólares.
b) exper: Un incremento en un año de experiencia, tiene un efecto de aumento en el salario de un 0.0701 dólares.
En cuanto a la hipótesis nula y alternativa, decimos que:

H0:\Bjfm=\Bjfn 
H1:\Bjfm≠\Bjfn 

Considerando que la significancia estádistica de las variables independientes son evaluadas de acuerdo con los intervalos de confianza, en donde el límite inferior es 0,025 y el superior 0,975, se puede concluir que:

a) educ: es estadisticamente signicativo, por lo que, se rechaza la hipótesis nula, ya que, estos límites no se encuentran dentro del 0.
b)exper: es estadisticamente signicativo, por lo que, se rechaza la hipótesis nula, ya que, estos límites no se encuentran dentro del 0.
Por lo tanto, las dos variables son estadisticamente signicativas, es decir, estas pueden afectar al salario (dependiente).
